### PR TITLE
Allow custom schema examples in OpenAPI format

### PR DESCRIPTION
**Pull request by mxck**
Hi! 
I added the ability to pass examples directly through the "x-examples" field (check this https://github.com/fastify/fastify-swagger/issues/535#issuecomment-1079928296)  and added documentation about examples in general.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by mxck** [comment]
Started to check the validity of the scheme through swagger editor and realized that it needs some fine-tuning. I'll reopen the PR later.

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mxck** [comment]
I found a error in example. 
And I need to add support for response and parameters objects.

File: README.md [hunk 1]
```diff
@@ -726,6 +726,125 @@ You can integration this plugin with ```@fastify/helmet``` with some little work
...
+          },
+          minItems: 1,
+        }
+      },
+      "x-examples": {
+        Cats: {
+          summary: "Feed cats",
+          description: 
+            "A longer **description** of the options with cats",
+          value: ["Tom", "Garfield", "Felix"]
```
Comment (mxck) [inline README.md:833]: Example should be the object

File: README.md [hunk 2]
```diff
@@ -726,6 +726,125 @@ You can integration this plugin with ```@fastify/helmet``` with some little work
...
+        }
+      },
+      "required": true
+    },
+    "responses": { "200": { "description": "Default Response" } }
+  }
+}
+```
+
+If you want to set your own names or add descriptions to the examples of schemas, you can use `x-examples` field to set examples in OpenAPI format:
```
Comment (Eomm) [inline README.md:809]:
```suggestion
If you want to set your own names or add descriptions to the examples of schemas, you can use `x-examples` field to set examples in [OpenAPI format](https://swagger.io/specification/#example-object):
```

**Review by Eomm — APPROVED** [review]

**Comment by mxck** [comment]
Please look again. I added support to params, response and body.

File: README.md [hunk 3]
```diff
@@ -726,6 +726,129 @@ You can integration this plugin with ```@fastify/helmet``` with some little work
...
+      properties: {
+        animals: {
+          type: 'array',
+          items: {
+            type: 'string'
+          },
+          minItems: 1,
+        }
+      },
+      "x-examples": {
```
Comment (mcollina) [inline README.md:828]: Could you make this example complete? I think we need a custom Ajv config, i.e. `ajv.addKeyword()` to make this work.

Comment (mxck) [reply README.md:839]: Done

**Review by mcollina — APPROVED** [review]
lgtm

**Review by climba03003 — APPROVED** [review]
LGTM.

**Pull request merged**